app/commands/git/sync.rb

```diff
@@ -0,0 +1,54 @@
+module Git
+  class Sync
+    include Mandate
+    initialize_with :track, :synced_to_git_sha
+
+    def call
+      update_git_repo!
+      sync! unless synced_to_head?
+    end
+
+    def sync!
+      raise NotImplementedError
+    end
+
+    def update_git_repo!
+      git_repo.update!
+    end
```

Comment:
You could probably inline this now. Katrina definitely would. I don't mind either way :)

Reply:
I prefer not to to keep everything at the same "level":

```ruby
def call
  update_git_repo!
  sync! unless synced_to_head?
end
```

to me reads better than:

```ruby
def call
  git_repo.update!
  sync! unless synced_to_head?
end
```

The dotting into the `git_repo` field to me seems to suggest more of an implementation detail than the first version.